maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1079,6 +1084,33 @@ private String getResumeFrom( List<MavenProject> mavenProjects, MavenProject fai
         return ":" + failedProject.getArtifactId();
     }
 
+    private boolean createResumeFromCacheFile( MavenExecutionResult result, String resumeFromProject )
+    {
+        String buildDirectory = result.getProject().getBuild().getDirectory();
+        File buildDirectoryFile = new File( buildDirectory );
+        boolean isBuildDirectoryPresent = ( buildDirectoryFile.exists() || buildDirectoryFile.mkdirs() );
+        if ( isBuildDirectoryPresent )
+        {
+            Path resumeFromCache = Paths.get( buildDirectory, "resume-from-cache" );
+            try
+            {
+                Files.write( resumeFromCache, resumeFromProject.getBytes() );
+                return true;
+            }
+            catch ( IOException e )
+            {
+                slf4jLogger.debug( "Failed writing last failed project to resume-from-cache file at: {}",
+                        resumeFromCache, e );
+                return false;
+            }
+        }
+        else
+        {
+            slf4jLogger.debug( "Failed creating build directory for root project to store resume-from-cache file" );
```

Comment:
Wouldn't it make sense to use a little more severe message? Most uses will not see them, so they will not know they are missing out on this great new feature :-)

Reply:
Sure, I was doubting about it. Anyway, I made them warnings now. @rfscholte do you think this is the proper logging severity?

Reply:
If we say that it should be possible to run a build without warnings, then this message should be info, because it doesn't effect the build itself, only the efficiency of the (next) build. (I know this  message is only shown after an error, but still...)

Reply:
I like that reasoning, I'll change it to info then!